src/python/pants/bin/remote_pants_runner.py

```diff
@@ -114,15 +118,19 @@ def run(self, start_time: float) -> ExitCode:
         pantsd_handle = self._client.maybe_launch()
         logger.debug(f"Connecting to pantsd on port {pantsd_handle.port}")
 
-        return self._connect_and_execute(pantsd_handle, start_time)
+        executor = GlobalOptions.create_py_executor(self._bootstrap_options.for_global_scope())
```

Comment:
Is the borrowed/ owned distinction introduced in https://github.com/pantsbuild/pants/pull/18166 still relevant - aka is there cleanup work to circle back to?

Reply:
Yes, it is. But the executor created via `create_py_executor` is always owned.